doc/book/app_server/luajit_getmetrics.rst

```diff
@@ -0,0 +1,441 @@
+
+Now see what happens if the table initialization is different:
+
+.. code-block:: lua
+
+    function f()
+      local table_new = require "table.new"
+      local oldm = misc.getmetrics()
+      local t = table_new(513, 0)
+      for i = 1, 513 do
+        t[i] = i
+      end
+      local newm = misc.getmetrics()
+      local diff = newm.gc_allocated - oldm.gc_allocated
+      print("diff = " .. diff)
+    end
+    f()
+
+The result will show that diff equals approximately 6000.
+
+.. _luajit_getmetrics_example_4:
+
+gc_steps_atomic and gc_steps_propagate
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The slope curves of gc_steps_* items can be used for tracking pressure on
+the garbage collector too.
+During long-running routines, gc_steps_* values will increase,
+but long times between ``gc_steps_atomic`` increases are a good sign.
+
+Also, increases in the ``gc_steps_propagate`` number can be used to
+estimate indirectly how many objects there are. These values also correlate with the
+garbage collector's
+`step multiplier <https://www.lua.org/manual/5.4/manual.html#2.5.1>`_.
+The number of incremental steps can grow, but
+one step can process a small number of objects. So these metrics should be
+considered when configuring the garbage collector.
+
+The following function takes a casual look whether an SQL statement causes much pressure:
+
+.. code-block:: lua
+
+    function f()
+      collectgarbage("collect")
+      local oldm = misc.getmetrics()
+      collectgarbage("collect")
+      box.execute([[DROP TABLE _vindex;]])
+      local newm = misc.getmetrics()
+      print("gc_steps_atomic = " .. newm.gc_steps_atomic - oldm.gc_steps_atomic)
+      print("gc_steps_finalize = " .. newm.gc_steps_finalize - oldm.gc_steps_finalize)
+      print("gc_steps_pause = " .. newm.gc_steps_pause - oldm.gc_steps_pause)
+      print("gc_steps_propagate = " .. newm.gc_steps_propagate - oldm.gc_steps_propagate)
+      print("gc_steps_sweep = " .. newm.gc_steps_sweep - oldm.gc_steps_sweep)
+    end
+    f()
+
+And the display will show that the gc_steps_* metrics are not significantly
+different from what they would be if the ``box.execute()`` was absent.
+
+.. _luajit_getmetrics_example_5:
+
+Example with jit_trace_num and jit_trace_abort
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Just-in-time compilers will "trace" code looking for opportunities to
+compile. ``jit_trace_abort`` can show how often there was a failed attempt
+(less is better), and ``jit_trace_num`` can show how many traces were
+generated since the last flush (usually more is better).
+
+The following function does not contain code that can cause trouble for LuaJIT:
+
+.. code-block:: lua
+
+    function f()
+      jit.flush()
+      for i = 1, 10 do collectgarbage("collect") end
+      local oldm = misc.getmetrics()    
+      collectgarbage("collect")
+      local sum = 0
+      for i = 1, 57 do
+        sum = sum + 57
+      end
+      for i = 1, 10 do collectgarbage("collect") end
```

Comment:
**** We do not, and if this was code for a product, then it would be rejected. But I wanted to call more than once, without a complex check that might distract the reader, when we're only trying to illustrate the effect of what follows.